src/Actions/helpers/ReduxWeb3Provider.js

```diff
@@ -40,6 +41,13 @@ class ReduxWeb3Provider extends EventEmitter {
 
     this.dispatch(RPCRequestStarted(payload))
 
+    GA.reportEvent({
```

Comment:
This seems like a lot of boilerplate to pepper into our redux action creators, etc.

Did you look for a way to do this as a redux middleware? For example, I could imagine having a pattern called "Analytics Reducers" which plug into a custom middleware and rather than reducing actions to state changes as typical redux reducers do, they reduce actions to analytics updates. Ideally we'd express these updates in a generic way so we could plug in other analytics providers apart from Google (e.g. InfluxDB)

Reply:
I could also see accomplishing this via redux sagas.

Reply:
Again, sagas will have to wait for another task.